corehq/util/doc_processor/sql.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+
+from django.conf import settings
+
+from corehq.util.doc_processor.interface import DocumentProvider
+from corehq.util.pagination import ResumableFunctionIterator, ArgsProvider
+
+
+class SqlModelArgsProvider(ArgsProvider):
+    def __init__(self, model_filter_attribute):
+        self.model_filter_attribute = model_filter_attribute
+        if not settings.USE_PARTITIONED_DATABASE:
+            self.db_list = [None]  # use the default database
+        else:
+            from corehq.sql_db.config import PartitionConfig
+            partition_config = PartitionConfig()
+            self.db_list = partition_config.get_form_processing_dbs()
```

Comment:
not really - it would require a settings file change and the only thing that could change without a large migration is the ordering.